sys/cbor/Makefile

```diff
@@ -2,6 +2,8 @@ MODULE = cbor
 
 ifneq ($(shell uname -s),Darwin)
 	CFLAGS += -D_XOPEN_SOURCE=600
+else
+	CFLAGS += -D_GNU_SOURCE=1
```

Comment:
Why non-Darvin only? Is OSX not using newlib for cross-compiling?

Reply:
Because this is not a OS X related issue, but gcc-arm-none-eabi libc (and maybe also avr-gcc, cannot test because my compiler is broken).

Reply:
But `uname - s` returns info about the *host* system, not the toolchain in use.

Reply:
True, let me check why it works then.

Reply:
The non darwin guard was to prevent multiple definition: https://github.com/RIOT-OS/RIOT/commit/e867d83e
However the test only made sense for native.

I also had the problem with arm-gcc `4.9.4`, but not with 6.3.1 anymore.

The new GNU_SOURCE you set is only done for Darwin (it is an ifneq). So it does not fixes linux + arm-gcc 4.9.4.